### PR TITLE
xDS retry policy should be considered present even if there are no supported policies in retry_on

### PR DESCRIPTION
**Pull request by donnadionne**
retry_on statuses) still takes precedence over virtual host level retry
policy.

Added a test to guard this case.

### CODE REVIEW

File: src/core/ext/xds/xds_api.cc [hunk 1]
```diff
@@ -1560,10 +1560,13 @@ grpc_error_handle RetryPolicyParse(
...
-  // return a policy if per_try_timeout is set even if retry_on specified no
-  // supported policies.
-  if (retry_to_return.retry_on.Empty()) return GRPC_ERROR_NONE;
+  if (retry_to_return.retry_on.Empty()) {
+    // we dont have any supported retry_on status, but we will still retrun an
+    // empty policy so that the per route policy takes precedence.  Returning a
+    // policy here also helps in the future where per_try_timeout is supported.
+    *retry = retry_to_return;
+    return GRPC_ERROR_NONE;
+  }
```
Comment (dapengzhang0) [inline src/core/ext/xds/xds_api.cc:1569]: Why not remove this block entirely and still perform validation for other fields?

Comment (markdroth) [reply src/core/ext/xds/xds_api.cc:1569]: I agree, I think it would make more sense to validate the rest of the policy.

**Review by markdroth — APPROVED** [review]
Looks good, other than Penn's comment, which I agree with.  Feel free to merge after fixing.

File: src/core/ext/xds/xds_api.cc (hunk 1 above)
Comment (donnadionne) [reply src/core/ext/xds/xds_api.cc]: removed block as suggested

**Review by dapengzhang0 — APPROVED** [review]

**Pull request merged**